resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -74,11 +74,7 @@ private[spark] class KubernetesClusterSchedulerBackend(
   private val executorCores = conf.getOption("spark.executor.cores").getOrElse("1")
 
   private implicit val requestExecutorContext = ExecutionContext.fromExecutorService(
-    Executors.newCachedThreadPool(
-      new ThreadFactoryBuilder()
-        .setDaemon(true)
-        .setNameFormat("kubernetes-executor-requests-%d")
-        .build))
+    ThreadUtils.newDaemonSingleThreadExecutor("kubernetes-executor-requests-%d"))
```

Comment:
any reason why you changed the executor service from CachedThreadPool to a SingleThreadExecutor ?
 
IIRC, ThreadUtil has a wrapper for newDaemonCachedThreadPool as well. Perhaps, you want that instead ?